### ISSUE TITLE
`check-param-names` should have a `contexts` option like the rest of the rules do

### ISSUE THREAD

**Issue by Zamiell** [gajus/eslint-plugin-jsdoc#897]
I believe that `check-param-names` should have a `contexts` option like the rest of the rules do.
That fact that it does not already have this seems to be an oversight.
Please implement this; it should be a simple fix!

This is needed in order to get a lint error for the following common situation:

```ts
/**
 * @param barr This is the description of bar. Oops, we misspelled "bar" as "barr".
 */
declare function foo(bar: number) {}
```

**Comment by brettz9** [comment]
I don't think we want to consider this as an oversight with there being no custom contexts for this rule and other function-only rules because checking all of the function contexts is intended and should be sufficient.

However, we were missing `TSDeclareFunction` as a default context to check, causing the sample code not to be reported as expected. That is now fixed.
